### PR TITLE
Remove deprecated methods on Decoder

### PR DESCRIPTION
**Pull request by fintelia**
Also removes `current_chunk` because it is no longer used elsewhere.

### CODE REVIEW

File: src/decoder/mod.rs [hunk 1]
```diff
@@ -1037,33 +1026,6 @@ impl<R: Read + Seek> Decoder<R> {
         Ok(u32::try_from(self.image().chunk_offsets.len())?)
     }
 
-    #[deprecated = "Use read_image instead"]
-    pub fn read_jpeg(&mut self) -> TiffResult<DecodingResult> {
-        self.read_image()
-    }
-
-    #[deprecated = "Use read_chunk_to_buffer instead"]
```
Comment (HeroicKatora) [inline src/decoder/mod.rs:1045]: Is `read_chunk` actually a good replacement yet? It has two additional arguments `chunk_index` and `output_width` to maintain.

File: src/decoder/mod.rs [hunk 2]
```diff
@@ -1139,30 +1101,6 @@ impl<R: Read + Seek> Decoder<R> {
         }
     }
 
-    /// Read a single strip from the image and return it as a Vector. This method does not return
-    /// information on the size of the strip read and so the caller must take care of the final
-    /// strip in the image as the data read may be less than the expected strip size. This interface
-    /// may change at a future date to reflect this.
-    #[deprecated = "Use read_chunk instead"]
-    pub fn read_strip(&mut self) -> TiffResult<DecodingResult> {
```
Comment (HeroicKatora) [inline src/decoder/mod.rs:1147]: A bit simpler, here it's only the `chunk_index` to maintain.

**Review by HeroicKatora — COMMENTED** [review]
> so I think that solves your concern?

Not really. My concern is that the replacement that the deprecations recommend don't appear to be (ergnomomic) replacements to me. I'd be _much_ gladder about the outright removal if we had a better migration story for downstream consumers.

**Comment by HeroicKatora** [comment]
Not that we shouldn't do it regardless of migration, the breaking change does make sense because the current tracking of state isn't that well-designed either. In terms of maintenace, definitely better. I'd just be glad about a plan for replacing it.

File: src/decoder/mod.rs (hunk 2 above)
Comment (fintelia) [reply src/decoder/mod.rs:1147]: I'd argue that it is _better_ to have the `chunk_index` get passed in by the user. It is useless for the user to get a chunk of image data without knowing which one it is, so they have to be tracking that information anyway. The new approach avoids the risk of the user's value getting out of sync with the library internal version

File: src/decoder/mod.rs (hunk 1 above)
Comment (fintelia) [reply src/decoder/mod.rs:1045]: This version of the method requires the user to allocated the buffer themselves. Which means that they have to know the `output_width` of their buffer either way, and the extra argument just gives a bit more flexibility (though looking at `read_chunk_to_buffer` we should probably add a doc comment actually explaining what the arguments mean)

File: src/decoder/mod.rs (hunk 2 above)
Comment (HeroicKatora) [reply src/decoder/mod.rs:1147]: A good point. My view was from the side of processing a complete image but even that keeps the index implicitly in an iterator. Very well then, let's just see if someone comes up with a more useful and appropriate abstraction in the new interface.

**Review by HeroicKatora — APPROVED** [review]

**Comment by HeroicKatora** [comment]
Quoting relevant information from the discussion:

> It is useless for the user to get a chunk of image data without knowing which one it is, so they have to be tracking that information anyway.

So, if anyone happens to run across this PR as the 'cause' of breakage then know that we are open to design ideas that reflect the above in a well-design interface.

**Pull request merged**